runtime/lua/vim/lsp.lua

```diff
@@ -2022,9 +2022,10 @@ end
 ---@param bufnr (integer) Buffer handle, or 0 for current.
 ---@param method (string) LSP method name
 ---@param params (table|nil) Parameters to send to the server
----@param callback (function) The callback to call when all requests are finished.
+---@param callback (fun(request_results: table<integer, {error: lsp.ResponseError, result: any}>))
```

Comment:
There is special syntax to make this work with vimdoc.

```
@param fun(request_results: table<integer, {error: lsp.ResponseError, result: any}>) (function) nlabla
```

Same with @return

You can always put `(vimdoctype)` in the comment, so after the type definition. That will then be used as the type in the vimdoc instead.

For more examples, check shared.lua

Reply:
Thanks for the info. Fixed.